HPCC-12464 Dynamic ESDL U/I

miguelvazq opened the pull request:
Incorporate a GUI for dynamic ESDL.

Signed-off by: Miguel Vazquez miguel.vazquez@lexisnexis.com

miguelvazq commented:
@GordonSmith Please review.

HPCCSmoketest commented:
Automated Smoketest 
Sha: 7c7befbb7ff7976ed93d028e19dd5a843312258b
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

miguelvazq commented:
FYI @JamesDeFabia

richardkchapman commented:
@GordonSmith

GordonSmith commented on esp/src/eclwatch/DynamicESDLDetailsWidget.js:
```diff
@@ -0,0 +1,313 @@
+
+        postCreate: function (args) {
+            var context = this;
+            this.inherited(arguments);
```
Very minor:  always cleaner to have the inherited call as the first line item.

GordonSmith commented on esp/src/eclwatch/DynamicESDLDetailsWidget.js:
```diff
@@ -0,0 +1,313 @@
+            this.addDefinitionButton = new Button({
+                disabled: false,
+                style: "float:right;",
+                label: "Apply",
```
Translation

GordonSmith commented on esp/src/eclwatch/DynamicESDLDetailsWidget.js:
```diff
@@ -0,0 +1,313 @@
+                                    Config: "<Methods><Method name='" + response.GetESDLDefinitionResponse.Methods.Method[0].Name + "'/></Methods>"
+                                }
+                            }).then(function (publishresponse) {
+                                if (publishresponse.PublishESDLBindingResponse.status.Code === 0) {
```
Does this need an "exists" test before using it (there might have been an exception)?

GordonSmith commented on esp/src/eclwatch/DynamicESDLDetailsWidget.js:
```diff
@@ -0,0 +1,313 @@
+                if (!context.widget.Configuration.initalized) {
+                    context.widget.Configuration.init({
+                        Configuration: context.params
+                });
```
Minor: indentation

GordonSmith commented on esp/src/eclwatch/DynamicESDLMethodWidget.js:
```diff
@@ -0,0 +1,365 @@
+            this.saveBindingButton = new Button({
+                disabled: true,
+                label: "Save"
+            }).placeAt(this.openButton.domNode, "after");*/
```
Remove commented code.

GordonSmith commented on esp/src/eclwatch/DynamicESDLMethodWidget.js:
```diff
@@ -0,0 +1,365 @@
+                label: "Attribute",
+                placeholder: "Attribute",
+                style: "margin-bottom: 10px;"
+
```
Blank line

GordonSmith commented on esp/src/eclwatch/DynamicESDLMethodWidget.js:
```diff
@@ -0,0 +1,365 @@
+                label: "Save"
+            }).placeAt(this.openButton.domNode, "after");*/
+            this.attributeTextBox = new TextBox({
+                label: "Attribute",
```
Translation?

GordonSmith commented on esp/src/eclwatch/DynamicESDLMethodWidget.js:
```diff
@@ -0,0 +1,365 @@
+
+            }).placeAt(this.dialog);
+            this.valueTextBox = new TextBox({
+                label: "Value",
```
Translation?

GordonSmith commented on esp/src/eclwatch/DynamicESDLMethodWidget.js:
```diff
@@ -0,0 +1,365 @@
+                onClick: function (val) {
+                    context.dialog.show();
+                },
+                label: "Add Attribute(s)"
```
Translation?

GordonSmith commented on esp/src/eclwatch/DynamicESDLMethodWidget.js:
```diff
@@ -0,0 +1,365 @@
+
+            this.store.getChildren = function (parent, options) {
+               var retVal =  this.query({__hpcc_parentName: parent.__hpcc_id}, options);
+               return retVal;
```
Minor: no need for retVal;

miguelvazq replied:
Tested and children do not show if removed.

GordonSmith replied:
I meant ```return this.query({__hpcc_parentName: parent.__hpcc_id}, options);``` (minor)

GordonSmith commented on esp/src/eclwatch/DynamicESDLMethodWidget.js:
```diff
@@ -0,0 +1,365 @@
+        refreshActionState: function (event) {
+            var context = this;
+            var selection = this.grid.getSelected();
+            var hasSelection = false;
```
Not used

GordonSmith commented on esp/src/eclwatch/DynamicESDLMethodWidget.js:
```diff
@@ -0,0 +1,365 @@
+            var hasParent = false;
+
+            for (var i = 0; i < selection.length; ++i) {
+                hasSelection = true;
```
Not used

GordonSmith commented on esp/src/eclwatch/DynamicESDLQueryWidget.js:
```diff
@@ -0,0 +1,152 @@
+
+            this.store.getChildren = function (parent, options) {
+               var retVal =  this.query({__hpcc_parentName: parent.__hpcc_id}, options);
+               return retVal;
```
Minor: retval not needed

miguelvazq replied:
Tested and children do not show if removed.

GordonSmith replied:
I meant:  ```return this.query({__hpcc_parentName: parent.__hpcc_id}, options);``` (minor)

GordonSmith commented on esp/src/eclwatch/templates/DynamicESDLDetailsWidget.html:
```diff
@@ -0,0 +1,25 @@
+                    <p>${i18n.PleaseSelectADynamicESDLService}</p>
+                </div>
+            </div>
+            <div id="${id}_Binding" title="Binding" data-dojo-type="dijit.layout.BorderContainer">
```
Translation

GordonSmith commented on plugins/cassandra/cpp-driver:
```diff
@@ -1 +1 @@
-Subproject commit 468db35af8875930d424395808ddc9523c57c6f2
+Subproject commit d7bbad34db39a51f209c6fadd07c9ec3a0bb86b7
```
This should not be in your PR

Commit:
HPCC-12464 Dynamic ESDL U/I

Incorporate a GUI for dynamic ESDL.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest 
Sha: 73124a158d2d10c21c23257b3deac9af2ca6be92
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest 
Sha: b158824d557128cc5eef70d945725bf030fa9fe9
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
@miguelvazq This is yours...

GordonSmith commented on esp/src/eclwatch/TargetSelectClass.js:
```diff
@@ -417,6 +420,24 @@ define([
+        },
+
+       loadESDLDefinitions: function () {
+        var context = this;
```
Indentation

GordonSmith reviewed (approved):
Just minor + indentation comments

miguelvazq commented:
@GordonSmith updated.

HPCCSmoketest commented:
Automated Smoketest 
Sha: dacef99e538846b62d6c77fafa39ff3c70cdc800
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

miguelvazq commented:
Rebased to fix conflicts.

HPCCSmoketest commented:
Automated Smoketest 
Sha: 26da7f104d785c362162c03aa396d9f26f887db7
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith reviewed (approved):
Good to go.

richardkchapman commented:
Merged manually to resolve conflict
